clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -522,7 +523,30 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
         long updatedNow = this.time.milliseconds();
         List<ClientResponse> responses = new ArrayList<>();
         handleCompletedSends(responses, updatedNow);
-        handleCompletedReceives(responses, updatedNow);
+
+        try {
+            handleCompletedReceives(responses, updatedNow);
+        } catch (StaleMetadataException e) {
+            // close the connections to the traitor brokers upon this StaleMetadataException
+            List<String> currentNodes = new ArrayList<>();
+            for (Node node : this.metadataUpdater.fetchNodes()) {
+                currentNodes.add(node.idString());
+            }
+
```

Comment:
should we follow the procedure in "processDisconnection" to clean up all inflight requests that are targeted to these nodes?